### PR TITLE
[IPPInAppFeedback] UI: Payments banner placement

### PR DESCRIPTION
**Pull request by iamgabrielma**
Part of https://github.com/woocommerce/woocommerce-ios/issues/8530

## Description

This PR adds the in-app feedback banner in the Payments view, as well as the buttons to launch a survey and dismiss the banner. 

At the moment the survey is only dismissible when tapping on "dismiss" (not upon survey completion), and will show up again each time we navigate to the Payments view.

## TODO:
- [ ] Design has changed. Needs to be re-implemented.
- [ ] Dismissal and show me later/don't show again UI will be added on a different PR. Most probably using `FeatureAnnouncementCardView`

## Changes
- We reuse the existing `Notice` component to create a feedback notice banner in the Payments View.
- Extended Notice to accept a new `noticeTappedHandler` handler noticeTappedHandler, so we can use 2 different behaviours: When the notice is tapped (show survey), and when the action button is tapped (dismiss notice).
- Added a new `IPPFeedback` case to `SurveyViewController` which will launch and manage the Survey launched via the tapping the notice
- Added a temporary testing survey link ( `https://automattic.survey.fm/woo-app-ipp-in-app-feedback-testing` ), this will dynamically change in future PRs based on certain conditions.

## Testing instructions
1. Go to Menu > Payments > See the notice on the bottom of the screen
2. Tap the notice, a WebView showing a survey form will shop up. Scroll down and tap "Finish Survey", a "Feedback Sent" view will be shown and you'll be redirected to the Payments page.
3. Navigate to a different View, then navigate to Menu > Payments again. 
4. Tap on "X" and see the banner disappear.

## Screenshots

| Feedback Banner | Give Feedback  | Feedback Sent  | 
|---|---|---|
|  |  |  |

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift [hunk 1]
```diff
@@ -334,6 +339,67 @@ private extension InPersonPaymentsMenuViewController {
...
+            actionButtons: [giveFeedbackAction, dismissFeedbackAction]
+        )
+        let topBannerView = TopBannerView(viewModel: viewModel)
+        topBannerView.translatesAutoresizingMaskIntoConstraints = false
+        return topBannerView
+    }
+
+    private func configureFeedbackBanner() {
+        topBannerView = createFeedbackBanner()
+        //tableView.updateHeaderHeight()
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift]:
```suggestion

```

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift [hunk 2]
```diff
@@ -334,6 +339,67 @@ private extension InPersonPaymentsMenuViewController {
...
+        showFeedbackBanner()
+    }
+
+    private func showFeedbackBanner() {
+        guard let topBannerView = topBannerView else { return }
+
+        // Configures header container view:
+        // TODO: Find the correct width and height
+        let width = Int(tableView.frame.width)
+        let height = 100
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift]: At the moment I'm hardcoding the banner height here as may change with the different banner content once we show different surveys. Will address this bit on a future PR, but happy to get any recommendation on how to deal with this on a better way.

Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift]: You can take a look at `CouponsListViewController` to check how the banner is added. If I see that correctly, we don't even need to specify a hardcoded height; it is rendered automatically thanks to the `updateHeaderHeight`  method in our `UITableView+HeaderFooterHelpers` extension.
I recommend looking at the used approach there to add the banner, not only for the height but in general to keep consistency between the two implementations.

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift (hunk 1 above)
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift]: Indeed we can remove that commented code

File: WooCommerce/Classes/System/WooConstants.swift [hunk 3]
```diff
@@ -155,8 +155,11 @@ extension WooConstants {
         ///
 #if DEBUG
         case inAppFeedback = "https://automattic.survey.fm/woo-app-general-feedback-test-survey"
+        case IPPinAppFeedback = "https://automattic.survey.fm/woo-app-ipp-in-app-feedback-testing"
```
Comment (toupper) [inline WooCommerce/Classes/System/WooConstants.swift]: Now that we have two inApp feedbacks, the naming looks a bit confusing to me. To make it clearer I would simply name it 
`generalFeedback` and
`IPPFeedback`
We can omit the `inApp` prefix, we know that the feedback is triggered from the app.

**Comment by toupper** [comment]
Thanks for opening this PR @iamgabrielma. I added a couple of comments, and since testing might change depending on how the comments are addressed, I will wait for that before testing.
Apart from that, it might be more convenient to name the branch after the issue we have, to make it easier to link it (`issue/8530-banner-placement`). It can even be easier to split that big issue into smaller ones, but that's more your opinion.

File: WooCommerce/Classes/System/WooConstants.swift (hunk 3 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/System/WooConstants.swift]: Definitely, changed these at https://github.com/woocommerce/woocommerce-ios/pull/8532/commits/34b894e3754b7f303e0911501f4a70a39c1ea54f and https://github.com/woocommerce/woocommerce-ios/pull/8532/commits/7a13d16506c96f3791ab6e0735c61b5998c7e56e

**Comment by iamgabrielma** [comment]
Thanks for the review @toupper !

Due to design changes on this feature I've redone the banner to use a `Notice` instead, I updated the PR description, testing instructions, and screenshots with the new information. Here's also the design proposal, and how it looks right now when building the PR:

|  Design proposal | Current design  | 
|---|---|
| Screenshot 2023-01-10 at 13 21 50 |  |

If this seems a good approach, I'll polish the UI further, as I may need to create a new Notice component that looks closer to what we need. I found other components in the codebase that may be helpful, like `PermanentNotice`, but while that one resolves some of the problems of using `Notice`, it also brings new ones. Happy to get any pointer if there's a better ones to use, or a better way to handle these.

> it might be more convenient to name the branch after the issue we have, to make it easier to link it (issue/8530-banner-placement). It can even be easier to split that big issue into smaller ones, but that's more your opinion.

Yes, that was a mistake on my end when pushing the branch 😅 , the [rest of issues](https://github.com/woocommerce/woocommerce-ios/issues/8530) are labeled and tracked in a better way.

**Comment by iamgabrielma** [comment]
Closing this issue due to modifications to scope ( pdfdoF-20T-p2 ), as we won't be showing this banner anymore in the Payments view.

**Pull request closed without merging**